Third-party-libraries.md

```diff
@@ -0,0 +1,255 @@
+* [Unmanaged development dependencies](#unmanaged-development-dependencies)
+* [Managed dependencies](#managed-dependencies)
+  * [`start.py`](#startpy)
+  * [`install_third_party_libs.py`](#install_third_party_libspy)
+    * [Python packages](#python-packages)
+    * [Node modules](#node-modules)
+    * [Protobuf](#protobuf)
+  * [`install_third_party.py`](#install_third_partypy)
+    * [Manifest files](#manifest-files)
+    * [Redis and Elasticsearch](#redis-and-elasticsearch)
+  * [`install_backend_python_libs.py`](#install_backend_python_libspy)
+* [Upgrade dependencies](#upgrade-dependencies)
+  * [Upgrade backend libraries](#upgrade-backend-libraries)
+    * [Upgrade production, backend libraries installed using pip](#upgrade-production-backend-libraries-installed-using-pip)
+    * [Upgrade development, backend libraries](#upgrade-development-backend-libraries)
+  * [Upgrade frontend libraries](#upgrade-frontend-libraries)
+    * [Upgrade frontend libraries installed from npm](#upgrade-frontend-libraries-installed-from-npm)
+    * [Upgrade frontend libraries not installed from npm](#upgrade-frontend-libraries-not-installed-from-npm)
+* [Add dependencies](#add-dependencies)
+  * [Add a backend library](#add-a-backend-library)
+    * [Add a backend, production library from pip](#add-a-backend-production-library-from-pip)
+    * [Add a backend, development library from pip](#add-a-backend-development-library-from-pip)
+    * [Add a library not installable from pip](#add-a-library-not-installable-from-pip)
+  * [Add a frontend library](#add-a-frontend-library)
+    * [Add a frontend library from npm](#add-a-frontend-library-from-npm)
+    * [Add a frontend library from outside npm](#add-a-frontend-library-from-outside-npm)
+
+## Introduction
+
+At Oppia we have a lot of dependencies, and we have many different ways of managing them. Some of these dependencies are required for our production application, while others are only used for development. Below, we will discuss how all these dependencies are managed.
+
+## Unmanaged development dependencies
+
+Oppia has some development dependencies that the user has to manage themselves. These are:
+
+* Python 3.7
+* Google Chrome
+* git
+* Tools commonly found on Linux and MacOS systems like bash
+
+In our [[setup instructions|Installing-Oppia]], we walk developers through installing these.
+
+## Managed dependencies
+
+The rest of Oppia's development and production dependencies are managed by the Oppia code. These dependencies are installed to the following places:
+
+* `oppia_tools/` stores development tools that aren't used in production. This folder is a sibling of the repository root (`oppia/`), so if you are in the root of the repository, this folder is at `../oppia_tools/`. However, we'll use `oppia_tools/` here for simplicity.
+* `third_party/python_libs` stores our production Python dependencies.
+* `third_party/static` stores some of our production Node.js dependencies.
```

Comment:
Should have a clearer statement of the difference between third_party/static and node_modules.

Is it the case that the former is deployed directly whereas the latter is not? (Might need to check with @vojtechjelinek.)

Reply:
static is compiled into one file by us, for node_modules it is done by webpack

Reply:
done